components/game_table/valorant/game_table_character_custom.lua

```diff
@@ -0,0 +1,174 @@
+		return mw.html.create('th'):css('max-width', width):node(text)
+	end
+
+	local config = self.config
+
+	local nodes = Array.append({},
+		makeHeaderCell('Date', '100px'),
+		config.showTier and makeHeaderCell('Tier', '70px') or nil,
+		config.showType and makeHeaderCell('Type', '70px') or nil,
+		config.showIcon and makeHeaderCell(nil, '25px'):addClass('unsortable') or nil,
+		makeHeaderCell('Tournament'),
+		makeHeaderCell('Map'),
+		(config.showResult and config.mode == Opponent.solo) and makeHeaderCell('') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('K') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('D') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('A') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('Ratio') or nil,
+		config.showResult and makeHeaderCell('Picks'):addClass('unsortable') or nil,
+		config.showResult and makeHeaderCell(nil, '80px') or nil,
+		config.showResult and makeHeaderCell('Score') or nil,
+		config.showResult and makeHeaderCell(nil, '80px') or nil,
+		config.showResult and makeHeaderCell('vs. Picks'):addClass('unsortable') or nil,
+		config.showLength and makeHeaderCell('Length') or nil,
+		config.showVod and makeHeaderCell('VOD', '60px') or nil
+	)
+
+	local header = mw.html.create('tr')
+	Array.forEach(nodes, function (node)
+		header:node(node)
+	end)
+
+	return header
+end
+
+---@param participant table
+---@return number?
+function CustomCharacterGameTable:_getRatio(participant)
+	local kills = tonumber(participant.kills) or 0
+	local deaths = tonumber(participant.deaths) or 0
+	if deaths == 0 then
+		return nil
+	end
+
+	return MathUtil.round(kills / deaths, 1)
+end
+
+---@param match GameTableMatch
+---@param game CharacterGameTableGame
+---@return Html?
+function CustomCharacterGameTable:displayGame(match, game)
+	local makeCell = function (text)
+		return mw.html.create('td'):node(text)
+	end
+
+	local makeIcon = function (character)
+		if not character then return nil end
+		return mw.html.create('td')
+			:node(CharacterIcon.Icon{character = character, size = self.config.iconSize, date = game.date})
+	end
+
+	local opponent = match.result.opponent
+	local opponentVs = match.result.vs
+	if self.isCharacterTable then
+		local pickedBy = game.pickedBy
+		---@cast pickedBy -nil
+		if pickedBy == 2 then
+			opponent, opponentVs = opponentVs, opponent
+		end
+	end
+
+	local node = mw.html.create()
+		:node(makeCell(Page.makeInternalLink(game.map)))
+
+	if self.config.mode ~= Opponent.team then
+		local participant = game.participants[game.pickedBy .. '_' .. game.pickedByplayer]
+		if self.config.mode == Opponent.solo then
+			local index = Array.indexOf(game.picks[game.pickedBy], function (pick)
+				return participant.agent == pick
+			end)
+			node:node(index > 0 and makeIcon(table.remove(game.picks[game.pickedBy], index)) or makeCell())
```

Comment:
Is `table.remove` necessary here, or would this suffice?
```suggestion
			node:node(index > 0 and makeIcon(game.picks[game.pickedBy][index]) or makeCell())
```

Reply:
Its needed because we want to replicate the old player table behaviour.We show the played character and then the 4 remaining chars